### DISCUSSION TITLE
Manual "one time" join request

### DISCUSSION THREAD

**Discussion by mlegrand2** [Lora-net/LoRaMac-node#1129] [Q&A]
Hello, 

I use the code example supplied in the periodic-uplink-lpp example.
My class A device is powered with a battery, so I try to reduce the power needed as low as possible. The device should operate for some years without any operation on it.
It works without any problem if the gateway is always alive and with a good radio coverage.
My problem is the following : when I use LmHandlerJoin( ), in case of no good LoRa coverage, the device try to make the join again and again, until the duty cycle limitation. This will empty the battery...
How can I do to make only one join request attempt?

Thank you for your help.

**Comment by MedadRufus** [comment]
You could modify APP_TX_DUTYCYCLE in run time, to adjust how often you want to do the join request. You could try every hour?

**Reply by mlegrand2** [reply]
Thank you for your answer.
The APP_TX_DUTYCYCLE is there to trigger the data tx frame. I already use a timer to trigger the data tx. This works and this is not the problem.
The problem is that I want to control the JOIN request, to limit the JOIN request rate to approx 1/hour.
Thank you for your help.

**Reply by mlegrand2** [reply]
Hello,
I see 2 solutions : 

Solution 1, manage the join request in main : 
- trigger the join request with LmHandlerJoin() from main
- disable LmHandlerJoin() in OnJoinRequest if Status == LORAMAC_HANDLER_ERROR

Solution 2, let the LoRaMac layers manage the join squence, and change the DC parameters :
- change the value of BACKOFF_DC_1_HOUR from 100 to 250
So it will activate the duty cycle limitation faster, to limit the amount of energy spent for join in case of no LoRa coverage.
